core/src/main/java/greencity/service/impl/TipsAndTricksServiceImpl.java

```diff
@@ -0,0 +1,167 @@
+import greencity.dto.PageableDto;
+import greencity.dto.tipsandtricks.AddTipsAndTricksDtoRequest;
+import greencity.dto.tipsandtricks.AddTipsAndTricksDtoResponse;
+import greencity.dto.tipsandtricks.TipsAndTricksDto;
+import greencity.entity.TipsAndTricks;
+import greencity.exception.exceptions.NotFoundException;
+import greencity.exception.exceptions.NotSavedException;
+import greencity.repository.TipsAndTricksRepo;
+import greencity.service.FileService;
+import greencity.service.NewsSubscriberService;
+import greencity.service.TipsAndTricksService;
+import greencity.service.TipsAndTricksTagsService;
+import greencity.service.UserService;
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.imageio.ImageIO;
+import lombok.RequiredArgsConstructor;
+import org.modelmapper.ModelMapper;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.cache.annotation.CacheEvict;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.mock.web.MockMultipartFile;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import static org.apache.commons.codec.binary.Base64.decodeBase64;
+
+@Service
+@RequiredArgsConstructor
+public class TipsAndTricksServiceImpl implements TipsAndTricksService {
+    @Value("${messaging.rabbit.email.topic}")
+    private String sendEmailTopic;
+
+    private final TipsAndTricksRepo tipsAndTricksRepo;
+
+    private final UserService userService;
+
+    private final ModelMapper modelMapper;
+
+    private final RabbitTemplate rabbitTemplate;
+
+    private final NewsSubscriberService newsSubscriberService;
+
+    private final TipsAndTricksTagsService tipsAndTricksTagsService;
+
+    private final FileService fileService;
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public AddTipsAndTricksDtoResponse save(AddTipsAndTricksDtoRequest addTipsAndTricksDtoRequest, MultipartFile image,
+                                            String email) {
+        TipsAndTricks toSave = modelMapper.map(addTipsAndTricksDtoRequest, TipsAndTricks.class);
+        toSave.setAuthor(userService.findByEmail(email));
+        if (addTipsAndTricksDtoRequest.getImage() != null) {
+            image = convertToMultipartImage(addTipsAndTricksDtoRequest.getImage());
+        }
+        if (image != null) {
+            toSave.setImagePath(fileService.upload(image).toString());
+        }
+        toSave.setTipsAndTricksTags(addTipsAndTricksDtoRequest.getTipsAndTricksTags()
+            .stream()
+            .map(tipsAndTricksTagsService::findByName)
```

Comment:
Is it possible to get all tipsandtrciks by one call to DB ?

Reply:
Added such method to the repo